public/js/products.js

```diff
@@ -1,85 +1,70 @@
-(function(item, cartCountElement) {
-    [].forEach.call(item, element => {
-      element.addEventListener('click', event => {
-        event.preventDefault();
+function cartAPI(url, productId) {
+  return fetch(url, {
+    method: "POST",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json"
+    },
+    body: JSON.stringify({ productId })
+  });
+}
+
+(function(cartCountElem, productList, isMatchProduct) {

```

Comment:
There is no point in having immediately invoking self executing functions (IIFE) for every functionality, you could have this only at application or module level.